common/src/main/kotlin/com/cosmotech/api/utils/SecurityUtils.kt

```diff
@@ -10,6 +11,15 @@ fun getCurrentAuthentication(): Authentication? = SecurityContextHolder.getConte
 
 fun getCurrentUserName(): String? = getCurrentAuthentication()?.name
 
+fun getCurrentUserUPN(): String? =
+    (getCurrentAuthentication()?.principal as? AADOAuth2AuthenticatedPrincipal)?.attributes
+        ?.getOrDefault("upn", null) as?
+        String?
```

Comment:
The cast to `AADOAuth2AuthenticatedPrincipal` is too Azure-specific, which is not the goal of this generic file.
I suggest using its `org.springframework.security.oauth2.core.OAuth2AuthenticatedPrincipal` base interface methods instead.
```suggestion
(getCurrentAuthentication()?.principal as? OAuth2AuthenticatedPrincipal)?.getAttribute<String>("upn")
```
Or we could consider moving this to the dedicated `com.cosmotech.api.azure` package.